text/0000-compiler-team-trusted-contrib-maintainer.md

```diff
@@ -0,0 +1,401 @@
+=======================
+[guide-level-explanation]: #guide-level-explanation
+
+Contributors start without any particular privileges, permissions or
+responsibilities and can contribute whatever they'd like. Contributors can
+progress to [Team Members][team-members] and then [Maintainers][maintainers].
+
+Team Members
+------------
+[team-members]: #team-members
+
+Being able to grant permissions to trusted contributors quickly is beneficial to
+enable them to contribute to the project more efficiently and review and approve
+work of their collaborators.
+
+Any contributor can request to become a team members by contacting the
+compiler team's leads, or current maintainers and team members can
+nominate a contributor. Team leads will check for a reasonable contribution
+history, and will check if the current team have any serious concerns related to
+contributor conduct (waiting approximately one week).
+
+When evaluating a candidate's contribution history, length of time and
+consistency of contributions and interactions with other contributors and
+maintainers will be taken into account. It is important to note that many kinds
+of contributions will be considered such as code contributions, helping with
+issue triage and bisection, running meetings and creating minutes, documentation
+contributions for rustc internals or the [Compiler Development Guide], etc.
+
+Team member is a mix of RFC 2689's "working group participant" and
+"compiler team contributor" roles. It is explicitly intended to be granted more
+liberally to contributors who have demonstrated competence and trustworthiness,
+for whom they would be able to work more effectively with these permissions and
+can be trusted to use them responsibly. Team members do not need to
+have experience with most of the compiler, and can be specialised to specific
+subsystems of the compiler.
+
+Team members are granted *r+*, *try*, *triagebot*, *rustc-perf*, and
+*crater* permissions; *organization membership*; and *dev desktop* access.
+Team members are considered members of the Rust project as a whole, and
+are automatically eligible for any benefits that incurs (e.g. invitations to
+meetups of project members). As representatives of the Rust project, trusted
+contributors are expected to obey not just the letter of the
+[Code of Conduct][coc] but its spirit.
+
+Team members can choose to take on additional responsibilities, such as
+those listed in the [responsibilities][responsibilities] section. Participating
+in the team's review rotation is encouraged.
+
+If a team member becomes inactive (the contributor's prior contributions and
+other interactions with the project cease) for longer than a year, the team
+member will be moved into alumni status. At any point in the future, they can
+ask to be re-instated at the team member level if they desire.
+
+[Compiler Development Guide]: https://rustc-dev-guide.rust-lang.org/
+
+Maintainers
+-----------
+[maintainers]: #maintainers
```

Comment:
I don't know where but would it make sense to point out somewhere that being maintainers is completely orthogonal to team [rolling leadership](https://rust-lang.github.io/rfcs/3262-compiler-team-rolling-leads.html#term-length-and-leader-selection)?

A maintainer can also never be a team lead and there is no requirement/expectations in this regard (IIUC).

Reply:
I've added a brief section on team leads pointing at that RFC.